### PR TITLE
Fix rendering issues

### PR DESCRIPTION
**Pull request by sorskoot**
- Fix issues with the context being null sometimes.
- Correct the way the material context and theme context are used for text.

### CODE REVIEW

File: js/components.tsx [hunk 1]
```diff
@@ -137,10 +138,14 @@ export const Text = forwardRef<
     >
 >((props, ref) => {
     const context = useContext(MaterialContext);
+    const theme = useContext(ThemeContext);
     const mat = props.material ?? useMemo(() => context.textMaterial?.clone(), []);
     if (mat) {
         (mat as unknown as FlatMaterial).setColor(
-            parseColor(props.color ?? 'fff', tempColor)
+            parseColor(
+                props.color ?? theme.colors?.text ?? (mat as unknown as FlatMaterial).color,
```
Comment (Squareys) [inline js/components.tsx:146]:
```suggestion
                props.color ?? theme.colors?.text ?? ((props.material ?? context.textMaterial) as unknown as FlatMaterial).color,
```

Alright, this is a bit of a puzzler 😅 This fixes a very niche bug where someone might have had a color property set, but then based on some UI state reverts it back to the "default", which is the theme or the _original_ material (not the clone).

File: js/renderer.ts [hunk 2]
```diff
@@ -454,7 +454,6 @@ function applyToYogaNode(
         node.setHeight(h);
         node.setWidth(w);
     } else {
-        applyLayoutToSceneGraph(wrapper, ctx!, true);
```
Comment (Squareys) [inline js/renderer.ts:457]: Are you sure about this one? That is quite a wide-ranging change 🤔

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@wonderlandengine/react-ui",
-  "version": "0.1.0-dev.7",
+    "version": "0.1.0-dev.9",
```
Comment (Squareys) [inline package.json:3]: Formatting seems off here. Maybe format the file with the prettier-config package?

**Review by Squareys — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (sorskoot) [reply package.json]: I formatted the file correctly.

File: js/renderer.ts (hunk 2 above)
Comment (sorskoot) [reply js/renderer.ts:457]: This was causing the main issue of the UI failing. Ctx turned out to be undefined sometimes. I changed it so there is an `if` around it.

**Pull request merged**